src/Nethermind/Nethermind.Core/ChainLevelInfo.cs

```diff
@@ -97,12 +110,19 @@ public void InsertBlockInfo(Hash256 hash, BlockInfo blockInfo, bool setAsMain)
             }
 
             int index = foundIndex ?? blockInfos.Length - 1;
+            int beaconMainChainIndex;
 
             if (setAsMain)
             {
                 blockInfos[index] = blockInfos[0];
                 blockInfos[0] = blockInfo;
             }
+            // prioritise new beacon info from beacon sync over old fcu
+            else if (blockInfo.IsBeaconMainChain && (beaconMainChainIndex = FindBeaconMainChainIndex()) != NotFound)
```

Comment:
other way would be `bool TryFindBeaconMainChainIndex(out int beaconMainChainIndex)`, which would make usage easier